app/platform/fabric/utils/FabricUtils.js

```diff
@@ -134,15 +134,15 @@ async function generateDir() {
 async function generateBlockHash(header) {
 	const headerAsn = asn.define('headerAsn', function() {
 		this.seq().obj(
-			this.key('Number').octstr(),
+			this.key('Number').int(),
 			this.key('PreviousHash').octstr(),
 			this.key('DataHash').octstr()
 		);
 	});
 	logger.info('generateBlockHash', header.number.toString());
 	const output = headerAsn.encode(
 		{
-			Number: header.number.toString(),
+			Number: parseInt(header.number),
```

Comment:
Based on your patch, I did some changes in my local. With the following patch, current block hash of each block gets identical with the previous hash of the next block. Your thought? 
```
diff --git a/app/platform/fabric/utils/FabricUtils.js b/app/platform/fabric/utils/FabricUtils.js
index 664e115..e7e7f94 100644
--- a/app/platform/fabric/utils/FabricUtils.js
+++ b/app/platform/fabric/utils/FabricUtils.js
@@ -134,7 +134,7 @@ async function generateDir() {
 async function generateBlockHash(header) {
        const headerAsn = asn.define('headerAsn', function() {
                this.seq().obj(
-                       this.key('Number').octstr(),
+                       this.key('Number').int(),
                        this.key('PreviousHash').octstr(),
                        this.key('DataHash').octstr()
                );
@@ -142,9 +142,9 @@ async function generateBlockHash(header) {
        logger.info('generateBlockHash', header.number.toString());
        const output = headerAsn.encode(
                {
-                       Number: header.number.toString(),
-                       PreviousHash: header.previous_hash.toString('hex'),
-                       DataHash: header.data_hash.toString('hex')
+                       Number: parseInt(header.number.low),
+                       PreviousHash: header.previous_hash,
+                       DataHash: header.data_hash
                },
                'der'
        );
```